QEfficient/version.py

```diff
@@ -1,8 +0,0 @@
-# -----------------------------------------------------------------------------
-#
-# Copyright (c)  2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
-# SPDX-License-Identifier: BSD-3-Clause
-#
-# -----------------------------------------------------------------------------
-
-__version__ = "0.0.1.dev0"
```

Comment:
Umm, is there a purpose to do this?
I am thinking we can make version correspond to SDK release, like current version can be `0.0.1.dev0.1-16-0` (can be better)?
So, people easily know which SDK is supported by their qeff package?

Reply:
@ochougul , That is not a standard version specifier allowed as per [PyPA standards](https://packaging.python.org/en/latest/specifications/version-specifiers/). For versions, I suggest we do either of the following:
- Follow transformers version with SDK as local version identifiers. Eg: `4.41.2+sdk1.16.0`, `4.36.2+sdk1.15.0` etc., since we aim to be API-compatible with transformers.
- Follow SDK versioning 1.15.0, 1.16.0, etc.
- Follow our own version like `1.0.0`, `0.1.0.dev20240624` etc.

I'm leaning towards first versioning scheme. In any case, we should use a date for `dev` because otherwise pypi tools tend to skip the installation, as the versions will remain the same.

Reply:
Agree.
So, the final version will look like `0.0.1.dev20246+transformers4.41.0_sdk1.15.0`?
It's too long.
We can think of transformers as upstream project and SDK as downstream project, and we want to be API compatible with both.
And since, this package will always be built from source, we are not redistributing it, we can and a `_` between `transformers` and `sdk` versions in the local version, PyPA allows such usage IMO.

Do you have better way of doing this? cause the version is too long like this.
Maybe we can drop transformers version from local version and instead put a check for its version during `QEfficient` import. And throw error/warning if user has different version installed in her/his environment.

Reply:
i feel we should keep it small, i do not think there is issue in existing release version `0.0.1.dev0`, we can think on numbering it accordingly. but adding transformers version will make it hard. Also i do not think we need sdk versioning mapped to this, since we already have branches and tags. We can have release version specific to branches.
@irajagop , i think we can keep `dev` without date, that's what i have seen in couple of open source [projects](https://github.com/huggingface/transformers/blob/main/src/transformers/__init__.py#L21).

Reply:
Ideally in that case, we should have following checks inside `QEfficient.__init__` in respective branches.
1. Check transformers version compatibility, prompt the user about correct version and raise warning in case of mismatch
2. Check SDK version compatibility, prompt the user about correct version and raise warning in case of mismatch

Reply:
Yeah, I'm also okay with keeping the version smaller, since we're not packaging or distributing it. We can finalize it to be `0.0.1.dev0`.

1. For the transformers version compatibility, there should be no need of checking it at runtime, as `pip install` will make sure the correct version is installed. Further, `pip check` can be used to ensure that.
2. For SDK compatibility, we can a check in `setup.py`, since that is run during `pip install`. That will be a separate PR.

Reply:
People may not always install the package, and just use `QEfficient` by doing editable install.
In that case `setup.py` will be run only during initial setup, otherwise when I update the repo and we change the transformers/sdk version support, `setup.py` won't run.

But I also agree, that adding this check logic in `__init__` leads to this being runs everytime `QEfficient` is imported.
Which might be okay, as it will take few milliseconds at max to check this. OR even better approach can be-> we creating a function that checks both of these things, and make it available to users, they can call this function to check versions?

Reply:
Lets finalize on `0.0.1.dev0`, the user changing transformers after installation, doesn't address this. But the same goes for any package in the `setup.py`, lets not complicate and keep simple versioning, we will update based on SDK branch releases, thanks. Closing the thread.